src/js/debug.js

```diff
@@ -0,0 +1,73 @@
+require.scopes.debug = (function() {
+let cookieFields = ['name', 'domain', 'hostOnly', 'path', 'secure', 'httpOnly', 'sameSite', 'session', 'expirationDate'];
+
+function printCookies(domain) {
+  getCookies(domain, true, (cookie) => {
+    console.log('----------------');
+    console.log(JSON.stringify(cookie, null, 2));
+  });
+}
+
+function getCookies(domain, showRelated, cookieCallback) {
+  let out ={};
+  if (showRelated) {
+    domain = window.getBaseDomain(domain);
+  }
+  out[domain] = [];
+  return new Promise((resolve) => {
```

Comment:
The problem with Promises is that they come with gotchas ("swallowed" exceptions, for example) and extra complexity (what the heck is a "promise"?). Unless we actually have a problem Promises make much simpler to implement (let's say, multiple asynchronous callbacks that should all finish before we do something else), I think plain old callbacks are just fine. Excessive nesting is usually simple to resolve by naming and moving anonymous functions, as we've done in this very PR elsewhere.